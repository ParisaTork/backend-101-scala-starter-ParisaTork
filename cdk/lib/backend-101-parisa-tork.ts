import { join } from "path";
import { GuDistributionBucketParameter, GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import { App, validateNumber } from "aws-cdk-lib";
import { CfnInclude } from "aws-cdk-lib/cloudformation-include";
import {GuEc2App} from "@guardian/cdk";
import { AccessScope } from "@guardian/cdk/lib/constants";
import { InstanceType, InstanceClass, InstanceSize } from "aws-cdk-lib/aws-ec2";

export class Backend101ParisaTork extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);
    const bucket = GuDistributionBucketParameter.getInstance(this).valueAsString;
    const name = "sheep";
    const keyPrefix = `${this.stack}/${this.stage}/${name}`;
    const userData = `#!/bin/bash -ev
    aws s3 cp s3://${bucket}/${keyPrefix}/app.service /etc/systemd/system/${name}.service
    aws s3 cp s3://${bucket}/${keyPrefix}/hello-world.jar /hello-world.jar
    systemctl start ${name}
    `;
    new GuEc2App(this, {
      applicationPort: 9000,
      app: "sheep",
      access: { scope: AccessScope.PUBLIC },
      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MEDIUM),
      certificateProps:{
        domainName: "sheep.code.dev-gutools.com",
      },
      monitoringConfiguration: {
        noMonitoring: true,
      },
      scaling: {
        minimumInstances: 1,
      },
      userData: {
        distributable: {
          fileName: "app-name.deb",
          executionStatement: `dpkg -i /app-name/app-name.deb`,
        }
      },
    });
  }
  

}
