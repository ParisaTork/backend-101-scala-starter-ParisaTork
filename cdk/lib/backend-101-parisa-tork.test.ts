import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { Backend101ParisaTork } from "./backend-101-parisa-tork";

describe("The Backend101ParisaTork stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new Backend101ParisaTork(app, "Backend101ParisaTork", { stack: "deploy", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
