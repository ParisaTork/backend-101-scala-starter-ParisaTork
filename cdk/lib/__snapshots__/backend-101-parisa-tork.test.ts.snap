// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Backend101ParisaTork stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [],
    "gu:cdk:version": "TEST",
  },
}
`;
